toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -68,6 +68,16 @@ func main() {
 		logger.Log.Fatalf("unable to read toolchain manifest file '%s': %s", *toolchainManifest, err)
 	}
 
+	dependencyGraph.InitLookup()
+
+	/*If there is an existing runNode and then there is one more remote node, donot throw dup error, instead replace the run node with remote node*/
+	for _, n := range graph.NodesOf(dependencyGraph.Nodes()) {
+		pkgNode := n.(*pkggraph.PkgNode)
+		if pkgNode.Type == pkggraph.TypeRemote {
+			dependencyGraph.AddRemoteToLookup(pkgNode, true)
+		}
+	}
+
```

Comment:
The `initLookup` method is meant for internal usage of `PkgGraph` as the whole lookup table is part of the internal implementation, which the caller/user shouldn't need to know about. Please revert this method back to be private and remove it from here.

Also, I'm not sure, if we even need to initialize the lookup table explicitly. We're calling `lookupTable()` inside of `AddRemoteToLookup()` and `lookupTable()` already includes a lazy initialization of the lookup table.

Reply:
right, lookupTable is initialized when used inside AddRemoteToLookup().